packages/token-metadata/src/TokenFactoryStatic.ts

```diff
@@ -0,0 +1,56 @@
+import { TokenType, TokenSource, TokenStatic } from './types'
+
+export class TokenFactoryStatic {
+  public registry: TokenStatic[]
+  public tokensByDenom: Record<string, TokenStatic>
+  public tokensBySymbol: Record<string, TokenStatic[]>
+
+  constructor(registry: TokenStatic[]) {
+    this.registry = registry
+    this.tokensByDenom = registry.reduce((list, token) => {
+      const denom = token.denom.toLowerCase()
+
+      if (list[denom]) {
+        return list
+      }
+
+      return { ...list, [denom]: token }
+    }, {} as Record<string, TokenStatic>)
+
+    this.tokensBySymbol = registry.reduce((list, token) => {
+      const symbol = token.symbol.toLowerCase()
+
+      return { ...list, [symbol]: [...(list[symbol] || []), token] }
```

Comment:
What will happen here if we have assets with same symbol but different denom?

Reply:
This map will return an array of Tokens: `Record<string, TokenStatic[]>`
reducing the list of tokens to searched based on symbol, for example, using source or type in the getMetaBySymbol function